semver.md

```diff
@@ -14,6 +14,9 @@ Given a version number MAJOR.MINOR.PATCH, increment the:
 Additional labels for pre-release and build metadata are available as extensions
 to the MAJOR.MINOR.PATCH format.
 
+Versions starting with zero are special, and follow the above rules matching
+the pattern `0.MAJOR.MINOR` and `0.0.MAJOR`.
```

Comment:
I think the wording here (and throughout the document) is somewhat ambiguous in some cases - specifically "moving" the major and minor term between X, Y and Z depending on the amount of leading zeroes.

Questions that can arise from this:

1. Is version `0.3.0` the same version as `3.0.0`, or are they different?
2. Similarly, how about `0.3.1` vs `3.1.0`?
3. Is it valid to go from `0.4.0` to `1.0.0`, or must you instead bump to `5.0.0` directly, React-style?

To be clear, I don't have an opinion on the _answer_ to those questions, all I care about is that the spec is written in such a way that the questions can be answered by just reading the spec.

@dominikg proposed the following alternative terminology in another discussion:

> every change that increased the leftmost non-zero segment can contain a breaking change ( `0.0.1`->`0.0.2` can break, `0.1.0` -> `0.1.1` is stable,  `0.1.2` -> `0.2.0` can break and so on)

This turns it around, to instead of defining a major as the left-most non-zero number, it defines breaking changes instead. With that perspective `0.4.1` would be major 0, minor 4, patch 1, but also say that a change in the minor version would be breaking. IMO this wouldn't cause any of the ambiguity listed above.

I also feel like it follows the spirit of "let the spec follow real usage", because I think the majority of users that sees `0.4.1` would not write that as "major version 4" but rather "major version 0". But this is just a gut feeling, I could be entirely wrong on this.

Anyway, that's just _an idea_ for another way to phrase this whole change, there are many ways to clear up the ambiguity.

Reply:
I think you're right; that's a better framing.